rir/src/runtime/Code.cpp

```diff
@@ -61,6 +76,71 @@ unsigned Code::getSrcIdxAt(const Opcode* pc, bool allowMissing) const {
     return sidx;
 }
 
+Code* Code::deserialize(SEXP refTable, R_inpstream_t inp) {
+    size_t size = InInteger(inp);
+    Code* code = (Code*)malloc(size);
+    code->uid = UUID::deserialize(refTable, inp) ^ uidHash;
+    code->funInvocationCount = InInteger(inp);
+    code->src = InInteger(inp);
+    code->stackLength = InInteger(inp);
+    *const_cast<unsigned*>(&code->localsCount) = InInteger(inp);
+    *const_cast<unsigned*>(&code->bindingCacheSize) = InInteger(inp);
+    code->codeSize = InInteger(inp);
+    code->srcLength = InInteger(inp);
+    code->extraPoolSize = InInteger(inp);
+    SEXP extraPool = ReadItem(refTable, inp);
+    PROTECT(extraPool);
+
+    // Bytecode
+    BC::deserialize(refTable, inp, code->code(), code->codeSize, code);
+
+    // Srclist
+    for (unsigned i = 0; i < code->srcLength; i++) {
+        code->srclist()[i].pcOffset = InInteger(inp);
+        code->srclist()[i].srcIdx =
+            src_pool_add(globalContext(), ReadItem(refTable, inp));
+    }
+    SEXP store = Rf_allocVector(EXTERNALSXP, size);
+    memcpy(DATAPTR(store), code, size);
+    Code* old = code;
+    code = (Code*)DATAPTR(store);
+    delete old;
```

Comment:
Same here, `malloc` and `delete` mismatch

Reply:
`free` caused segfaults where the code body got corrupted, I have no idea how

Reply:
Well but this is UB: https://isocpp.org/wiki/faq/freestore-mgmt#mixing-malloc-and-delete